stubs/fanstatic/fanstatic/compiler.pyi

```diff
@@ -0,0 +1,121 @@
+from _typeshed import StrOrBytesPath
+from abc import abstractmethod
+from logging import Logger
+from subprocess import Popen
+from typing import Any, ClassVar
+from typing_extensions import Literal
+
+import setuptools.command.sdist
+from fanstatic.core import Resource
+
+logger: Logger
+
+class CompilerError(Exception): ...
+
+class Compiler:
+    @property
+    @abstractmethod
+    def name(self) -> str: ...
+    @property
+    @abstractmethod
+    def source_extension(self) -> str: ...
+    def __call__(self, resource: Resource, force: bool = False) -> None: ...
+    def process(self, source: StrOrBytesPath, target: StrOrBytesPath) -> Any: ...
+    def should_process(self, source: StrOrBytesPath, target: StrOrBytesPath) -> bool: ...
+    @property
+    @abstractmethod
+    def available(self) -> bool: ...
+    def source_path(self, resource: Resource) -> str | None: ...
+    def target_path(self, resource: Resource) -> str | None: ...
+
+class Minifier(Compiler):
+    @property
+    @abstractmethod
+    def name(self) -> str: ...
+    @property
+    @abstractmethod
+    def source_extension(self) -> str: ...
+    @property
+    @abstractmethod
+    def target_extension(self) -> str: ...
+    def source_to_target(self, resource: Resource) -> str: ...
+
+def compile_resources(argv: list[str] = ...) -> None: ...
+
+class sdist_compile(setuptools.command.sdist.sdist): ...
+
+class NullCompiler(Compiler):
+    name: ClassVar[Literal[""]]
+    source_extension = NotImplemented
```

Comment:
I've been under the impression that `NoReturn` should only be used for functions like `sys.exit`. I have used `NoReturn` in this way in the past and it was generally met with opposition, since the errors generated by it aren't very clear.

Ideally this would use the `@type_error` decorator, once that is a thing. But in this case `source_extension` actuall is `NotImplemented` at runtime, so that seems fine to me.